pkg/util/certificates.go

```diff
@@ -108,91 +111,147 @@ type CertificateFactory struct {
 	domain  string
 }
 
+type CertUpdates struct {
+	Certificate *x509.Certificate
+	Type        string
+	Reason      string
+}
+
 func NewCertificateFactory(kluster *v1.Kluster, store *v1.Certificates, domain string) *CertificateFactory {
 	return &CertificateFactory{kluster, store, domain}
 }
 
-func (cf *CertificateFactory) Ensure() error {
+func (cf *CertificateFactory) Ensure() (error, []CertUpdates) {
```

Comment:
Thanks for taking care! I have one final remark, can we reverse the order of the return values? In general in go its custom that the error is always returned as the last argument. Its very uncommon to deviate from this.

Reply:
@databus23 I changed return order of `Ensure()`.

Reply:
❤️

Reply:
Will merge as soon as the test is green. Thanks a lot